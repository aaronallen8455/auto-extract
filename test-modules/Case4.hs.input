module Case4 where

test = do
  x <- EXTRACT@e (do
    y <- readLn
    pure (y + 1))
  pure $ x + 3
