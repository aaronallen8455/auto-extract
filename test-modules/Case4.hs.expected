module Case4 where

test = do
  x <- e
  pure $ x + 3

e :: IO Integer
e = do
  y <- readLn
  pure (y + 1)
